libs/domain/product/variant-selector/variant-selector.component.ts

```diff
@@ -0,0 +1,179 @@
+            (value) => html`
+              <oryx-toggle-icon>
+                <input
+                  type="radio"
+                  placeholder="make a11y happy"
+                  .name=${key}
+                  .value=${value}
+                  ?checked=${this.isChecked(key, value)}
+                  ?disabled=${this.isDisabled(key, value, index)}
+                />
+                <span>${value}</span>
+              </oryx-toggle-icon>
+            `
+          )}</form>`
+      )}
+    </form>`;
+  }
+
+  protected isChecked(key: string, value: string) {
+    const product = this.$product();
+    return product?.attributes?.[key] === value;
+  }
+
+  /**
+   * Determines if a given attribute is disabled based on the product's variant attribute info.
+   */
+  protected isDisabled(
+    attributeKey: string,
+    attributeValue: string,
+    index: number
+  ): boolean {
+    const product = this.$product();
+
+    if (!product?.variants) {
+      return false;
+    }
+
+    // Extract all active variant attribute keys except the one at the given index
+    const allKeys = Object.keys(product.variantDefinition || {});
+    const activeKeys = allKeys.filter((_, i) => i !== index);
+
+    // Check if any variant matches the selected attribute values
+    const isVariantDisabled = !Object.values(product.variants).some(
+      (variant) => {
+        // Check if all selected attribute values (except the one at the index) match the variant
+        return (
+          activeKeys.every(
+            (key) => variant[key] === this.$product()?.attributes?.[key]
+          ) && variant[attributeKey] === attributeValue
+        );
+      }
+    );
+
+    return isVariantDisabled;
+  }
+
+  // protected isDisabled(
+  //   attributeKey: string,
+  //   attributeValue: string,
+  //   index: number
+  // ): boolean {
+  //   const product = this.$product();
+
+  //   if (!product?.variants) {
+  //     return false;
+  //   }
+
+  //   // Extract active variant attribute keys up to the given index
+  //   const activeKeys = Object.keys(product.variantDefinition || {}).slice(
+  //     0,
+  //     index
+  //   );
+
+  //   // Check if any variant matches the selected attribute values
+  //   const isVariantDisabled = !Object.values(product.variants).some(
+  //     (variant) => {
+  //       // Check if all selected attribute values match the variant
+  //       return (
+  //         activeKeys.every(
+  //           (key) => variant[key] === this.$product()?.attributes?.[key]
+  //         ) && variant[attributeKey] === attributeValue
+  //       );
+  //     }
+  //   );
+
+  //   return isVariantDisabled;
+  // }
+
+  protected handleVariantChange(e: Event) {
+    const form = e.currentTarget as HTMLFormElement;
+    const formData = new FormData(form);
+
+    const selectedAttributes: Record<string, string> = {};
+    formData.forEach((value, key) => {
+      selectedAttributes[key] = value as string;
+    });
```

Comment:
Can we just create state component variable with all selected options and add change method to the option so in this case we should not recalculate the whole form each time?